kernel/memory_structs/src/lib.rs

```diff
@@ -24,6 +26,53 @@ use paste::paste;
 use derive_more::*;
 use range_inclusive::{RangeInclusive, RangeInclusiveIterator};
 
+/// Numeric constants associated with different page types.
+pub const PAGE_4KB_SIZE: usize = 1 << 12;
+pub const PAGE_2MB_SIZE: usize = (1 << 12) * 512;
+pub const PAGE_1GB_SIZE: usize = (1 << 12) * 512 * 512;
+
+/// Enum used to indicate the size of a page or frame.
+#[derive(Debug)]
+pub enum MemChunkSize {
+    Normal4K,
+    Huge2M,
+    Huge1G,
+}
+
+/// Trait used to group the size marker structs that paging-related types are parameterized with.
```

Comment:
Your current docs describe how it's used, not really what it means. Suggestion below:
```suggestion
/// Trait used to represent the size of a page or frame, i.e., for normal or huge pages.
///
/// This is used to parameterize `Page`- and `Frame`-related types with a page size,
/// in order to define normal and huge pages in a generic manner.
```